packages/kbn-reporting/generate_csv/types/index.d.ts

```diff
@@ -0,0 +1,274 @@
+
+export const ILM_POLICY_NAME = 'kibana-reporting';
+
+// Usage counter types
+export const API_USAGE_COUNTER_TYPE = 'reportingApi';
+export const API_USAGE_ERROR_TYPE = 'reportingApiError';
+
+// Management UI route
+export const REPORTING_MANAGEMENT_HOME = '/app/management/insightsAndAlerting/reporting';
+
+export const REPORTING_REDIRECT_LOCATOR_STORE_KEY = '__REPORTING_REDIRECT_LOCATOR_STORE_KEY__';
+
+// Export Reporting Type Definitions
+export const CSV_REPORT_TYPE = 'CSV';
+export const CSV_REPORT_TYPE_V2 = 'csv_v2';
+
+export const PDF_REPORT_TYPE = 'printablePdf';
+export const PDF_REPORT_TYPE_V2 = 'printablePdfV2';
+
+export const PNG_REPORT_TYPE = 'PNG';
+export const PNG_REPORT_TYPE_V2 = 'pngV2';
+
+// Export Job Type Definitions
+export const CSV_JOB_TYPE = 'csv_searchsource';
+
+export const PDF_JOB_TYPE = 'printable_pdf';
+export const PDF_JOB_TYPE_V2 = 'printable_pdf_v2';
+
+export const PNG_JOB_TYPE = 'PNG';
+export const PNG_JOB_TYPE_V2 = 'PNGV2';
+
+/**
+ * A way to get the client side route for the reporting redirect app.
+ *
+ * TODO: Add a job ID and a locator to use so that we can redirect without expecting state to
+ * be injected to the page
+ */
+export const getRedirectAppPath = () => {
+  return '/app/reportingRedirect';
+};
+
+// Statuses
+export enum JOB_STATUSES {
+  PENDING = 'pending',
+  PROCESSING = 'processing',
+  COMPLETED = 'completed',
+  FAILED = 'failed',
+  WARNINGS = 'completed_with_warnings',
+}
+
+// Test Subjects
+export const REPORT_TABLE_ID = 'reportJobListing';
+export const REPORT_TABLE_ROW_ID = 'reportJobRow';
+
+// Job params require a `version` field as of 7.15.0. For older jobs set with
+// automation that have no version value in the job params, we assume the
+// intended version is 7.14.0
+export const UNVERSIONED_VERSION = '7.14.0';
+
+// hacky endpoint: download CSV without queueing a report
+// FIXME: find a way to make these endpoints "generic" instead of hardcoded, as are the queued report export types
+export const API_GENERATE_IMMEDIATE = `${API_BASE_URL_V1}/generate/immediate/csv_searchsource`;
+
+export const CsvConfig = schema.object({
+  checkForFormulas: schema.boolean({ defaultValue: true }),
+  escapeFormulaValues: schema.boolean({ defaultValue: false }),
+  enablePanelActionDownload: schema.boolean({ defaultValue: true }),
+  maxSizeBytes: schema.oneOf([schema.number(), schema.byteSize()], {
+    defaultValue: ByteSizeValue.parse('10mb'),
+  }),
+  useByteOrderMarkEncoding: schema.boolean({ defaultValue: false }),
+  scroll: schema.object({
+    duration: schema.string({
+      defaultValue: '30s', // this value is passed directly to ES, so string only format is preferred
+      validate(value) {
+        if (!/^[0-9]+(d|h|m|s|ms|micros|nanos)$/.test(value)) {
+          return 'must be a duration string';
+        }
+      },
+    }),
+    size: schema.number({ defaultValue: 500 }),
+  }),
+});
+
+export type CsvConfigType = typeof CsvConfig;
```

Comment:
This type is needed for CSV generation, but we can simplify the whole thing, not use `schema.object` at all and do:
```
export interface CsvConfig {
  checkForFormulas: boolean;
  escapeFormulaValues: boolean;
  maxSizeBytes: string | number;
  useByteOrderMarkEncoding: boolean;
  scroll: {
    duration: string;
    size: number;
  };
}

```

I removed the things we don't need to know about for CSV generation, like `enablePanelActionDownload`